src/IECore/MemoryIndexedIO.cpp

```diff
@@ -36,106 +36,195 @@
 #include "IECore/FileIndexedIO.h"
 #include "IECore/VectorTypedData.h"
 
+#include <iosfwd>
+#include <boost/iostreams/stream.hpp>
+#include <boost/iostreams/positioning.hpp>
+
 using namespace IECore;
 
 IE_CORE_DEFINERUNTIMETYPEDDESCRIPTION( MemoryIndexedIO )
 
-///////////////////////////////////////////////
+//////////////////////////////////////////////////////
 //
-// FileIndexedIO::StreamFile (begin)
+// FileIndexedIO::StreamFile subclass implementations
 //
-///////////////////////////////////////////////
+//////////////////////////////////////////////////////
 
-class MemoryIndexedIO::StreamFile : public StreamIndexedIO::StreamFile
+class MemoryIndexedIO::CharVectorDataDevice
```

Comment:
I think I'd be tempted to make this a CharVectorDevice that is initialised with a `vector<char> &`, just to make the code a bit less wordy, and to increase the chance that we might reuse it elsewhere at some point. Is that possible or is there something I'm missing?